lib/modules/platform/codecommit/iam-client.spec.ts

```diff
@@ -0,0 +1,69 @@
+import { GetUserCommand, IAMClient } from '@aws-sdk/client-iam';
+import { mockClient } from 'aws-sdk-client-mock';
+import { PLATFORM_BAD_CREDENTIALS } from '../../../constants/error-messages';
+
+describe('modules/platform/codecommit/iam-client', () => {
+  const iam = require('./iam-client');
+  const iamClient = mockClient(IAMClient);
+
+  beforeAll(() => {
+    iam.initIamClient('eu-east', {
+      accessKeyId: 'aaa',
+      secretAccessKey: 'bbb',
+    });
+  });
+
+  it('should throw in case of bad authentication', async () => {
+    iamClient.on(GetUserCommand).rejects(new Error(PLATFORM_BAD_CREDENTIALS));
+    let userArn;
+    try {
+      userArn = await iam.getUserArn();
+    } catch (err) {
+      //
+    }
```

Comment:
use async jest matchers instead. search tests for `.rejects` samples

Reply:
i tried that before it didnt work, now it's working fine, i changed all try/catch tests